src/tooltip/Tooltip.js

```diff
@@ -0,0 +1,215 @@
+      alignItems: 'center',
+      justifyContent: 'center',
+      flex: 1,
+      borderRadius: 10,
+      padding: 10,
+      ...containerStyle,
+    };
+  };
+
+  renderPointer = () => {
+    const { yOffset, xOffset, elementHeight, elementWidth } = this.state;
+    const { backgroundColor, pointerColor } = this.props;
+    const pastMiddleLine = yOffset > ScreenHeight / 2;
+
+    return (
+      <View
+        style={{
+          position: 'absolute',
+          top: pastMiddleLine ? yOffset - 13 : yOffset + elementHeight - 2,
+          left: xOffset + elementWidth / 2 - 7.5,
+        }}
+      >
+        <Triangle
+          style={{ borderBottomColor: pointerColor || backgroundColor }}
+          isDown={pastMiddleLine}
+        />
+      </View>
+    );
+  };
+  renderContent = withTooltip => {
+    const { popover, withPointer, toggleOnPress, highlightColor } = this.props;
+
+    if (!withTooltip)
+      return this.wrapWithPress(toggleOnPress, this.props.children);
+
+    const { yOffset, xOffset } = this.state;
+    return (
+      <React.Fragment>
+        <View
+          style={{
+            position: 'absolute',
+            top: yOffset,
+            left: xOffset,
+            backgroundColor: highlightColor,
+            overflow: 'visible',
+          }}
+        >
+          {this.props.children}
+        </View>
+        {withPointer && this.renderPointer()}
+        <View style={{ ...this.getTooltipStyle() }}>{popover}</View>
+      </React.Fragment>
+    );
+  };
+
+  componentDidMount() {
+    // wait to compute onLayout values.
+    setTimeout(this.getElementPosition, 500);
+  }
+
+  getElementPosition = () => {
+    if (this.renderedElement) {
+      NativeMethodsMixin.measureInWindow.call(
+        this.renderedElement,
+        (x, y, width, height) => {
+          this.setState({
+            xOffset: x,
+            yOffset: y,
+            elementWidth: width,
+            elementHeight: height,
+          });
+        }
+      );
+    }
+  };
+
+  render() {
+    const { isVisible } = this.state;
+    const { onClose, withOverlay, onOpen } = this.props;
+
+    return (
+      <View collapsable={false} ref={e => (this.renderedElement = e)}>
+        {this.renderContent(false)}
+        <Modal
+          animationType="fade"
+          visible={isVisible}
+          transparent
+          onDismiss={onClose}
+          onShow={onOpen}
+          onRequestClose={onClose}
+        >
+          <TouchableOpacity
+            style={styles.container(withOverlay)}
+            onPress={this.toggleTooltip}
+            activeOpacity={1}
+          >
+            {this.renderContent(true)}
+          </TouchableOpacity>
+        </Modal>
+      </View>
+    );
+  }
+}
+
+Tooltip.propTypes = {
+  children: PropTypes.element,
+  withPointer: PropTypes.bool,
+  popover: PropTypes.element,
+  toggleOnPress: PropTypes.bool,
+  height: PropTypes.number,
+  width: PropTypes.number,
+  containerStyle: PropTypes.any,
```

Comment:
Use the style property from `ViewPropTypes`. See https://github.com/react-native-training/react-native-elements/blob/master/src/avatar/Avatar.js#L238 for example